staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go

```diff
@@ -353,8 +353,13 @@ func TestProgressNotify(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Watch failed: %v", err)
 	}
-	result := &example.Pod{ObjectMeta: metav1.ObjectMeta{ResourceVersion: out.ResourceVersion}}
-	testCheckResult(t, watch.Bookmark, w, result)
+	testCheckResultFunc(t, watch.Bookmark, w, func(object runtime.Object) error {
```

Comment:
this change is weakening the check, since it is no longer verifying it gets an object containing only the resource version